packages/odyssey-react-mui/src/Breadcrumbs.tsx

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  ButtonBase,
+  Menu,
+  MenuItem,
+  Breadcrumbs as MuiBreadcrumbs,
+} from "@mui/material";
+import { ReactElement, createContext, memo, useContext, useState } from "react";
+import { GroupIcon, HomeIcon, UserIcon } from "./icons.generated";
+import { Typography } from "./Typography";
+import { useTranslation } from "react-i18next";
+
+export type BreadcrumbProps = {
+  children?: string;
+  href: string;
+  isCurrent?: boolean;
+  startIcon?: null | ReactElement<typeof UserIcon | typeof GroupIcon>;
+};
+
+export type BreadcrumbsProps = {
+  children: ReactElement<typeof Breadcrumb>[];
+  homeHref?: string;
+  maxItemsCount?: number;
+};
+
+const BreadcrumbContext = createContext(false);
+
+export const Breadcrumb = ({
+  children,
+  href,
+  isCurrent,
+  startIcon,
+}: BreadcrumbProps) => {
+  const isInsideMenu = useContext(BreadcrumbContext);
+
+  const breadcrumbContent = (
+    <>
+      {startIcon}
+      {children}
+    </>
+  );
+
+  if (isInsideMenu) {
+    return <MenuItem href={href}>{breadcrumbContent}</MenuItem>;
+  }
+
+  if (isCurrent) {
+    return <Typography>{breadcrumbContent}</Typography>;
+  }
+
+  return <ButtonBase href={href}>{breadcrumbContent}</ButtonBase>;
+};
+
+const BreadcrumbList = ({
+  children,
+  homeHref,
+  maxItemsCount = children.length,
+}: BreadcrumbsProps) => {
```

Comment:
I don't like referencing other props in default values of other props because it depends on prop ordering, and we want to alphabetize prop names. It _only_ just happens to work here.

We should do this math in the component itself with a new name. Like we could rename this props to `maxItemsCount: maxItemsCountOverride` and use the name `const maxItemsCount = maxItemsCountOverride ?? children.length` in the component.